app/models/host/base.rb

```diff
@@ -99,13 +99,8 @@ def self.attributes_protected_by_default
       super - [ inheritance_column ]
     end
 
-    def self.import_host_and_facts(json)
-      # noop, overridden by STI descendants
-      [self, true]
-    end
-
```

Comment:
This will likely break discovery plugin. @eLobato @domcleal before merging please make sure we have a PR pending for discovery plugin.